README.md
# dcLatestVersionsLight
# README EN

## WHAT IS DCLATESTVERSIONSLIGHT ?

"Dotclear's latest versions light" is a plugin for the open-source 
web publishing software called __Dotclear__.

Simply show administrators last versions of nightly build of Dotclear:
_(stable, testings and unstable)_

## REQUIREMENTS

 dcLatestVersionsLight requires: 

  * permissions to user admin preferences
  * Dotclear 2.9
  * tested on php 5.6.40

 Dotclear versions tested:
  * Dotclear 2.9 to 2.13: php 5.6.40
  * Dotclear 2.14 to 2.15: php 7.3.5
  * Dotclear 2.16+: php 7.4.1

## USAGE

First install dcLatestVersionsLight, manualy from a zip package.
__(See Dotclear's documentation to know how do this)__

Add and configure dcLatestVersionsLight from plugin or "user preferences" interfaces.
You can add a dashboard item by enable it from your dashboard preferences.

---

# LISEZ MOI FR

## QU'EST-CE QUE DCLATESTVERSIONSLIGHT ?

"Dotclear's latest versions light" est un plugin pour 
le moteur de blog open-source __Dotclear__.

Simply show administrators last versions of nightly build of Dotclear.
Il montre simplement les dernières "nightly" versions de Dotclear:
_(stable, testings and unstable)_


## NÉCESSAIRE

 dcLatestVersionsLight demande: 

  * que vous ayez la permission admin pour modifier user preferences
  * Dotclear 2.9
  * testé sous php 5.6.40

 Dotclear versions testées:
  * Dotclear 2.9 to 2.13: php 5.6.40
  * Dotclear 2.14 to 2.15: php 7.3.5
  * Dotclear 2.16+: php 7.4.1

## USAGE

D'abord installer dcLatestVersionsLight, directement depuis un package zip.
(Voir la documentation Dotclear pour savoir comment faire)

Ajouter et configurer dcLatestVersionsLight depuis l'interface admin du plugin ou sous l'interface "préférences utilisateur".
Vous pouvez ajouter l'item sur sur votre tableau de bord en l'activant depuis l'interface préférences utilisateur.
You can add a dashboard item by enable it from your dashboard preferences.

----

### Admin dashboard view
![Screenshot](https://user-images.githubusercontent.com/62995223/90084038-86ddf900-dd14-11ea-85f7-e22417e300fb.jpg)

### Plugin Interface
![Screenshot](https://user-images.githubusercontent.com/62995223/90084041-880f2600-dd14-11ea-8b90-9aa9f2cfad7d.jpg)

### User Preferences menu
![Screenshot](https://user-images.githubusercontent.com/62995223/90084042-88a7bc80-dd14-11ea-977d-d5104a8f7e5c.jpg)


CHANGELOG
Change log
**********
# Changelog
All notable changes to this project will be documented in this file.

## information: this module work on:
	### Dotclear 2.9+ tests
		#### Dotclear versions & php versions see forum -> "Versions Dotclear & versions php", https://forum.dotclear.org/viewtopic.php?pid=344999
	### php versions tests :
		#### php 5.6.40
		#### php 7.3.5
	### Dotclear upload & information: https://plugins.dotaddict.org/dc2/details/dcLatestVersionsLight
	### Dotclear support (fr): https://forum.dotclear.org/viewtopic.php?id=49826
	### Github upload & support: https://github.com/nanart59/dcLatestVersionsLight

## [released]
    ### Dotclear version
		### Dotclear version 2.19 need php >= 7.3
		### french translate
## [unreleased]


## this module dcLatestVersionsLight
====================================
## [1.17] - 2021/04/12
    ### Change
        #### Dotclear version 2.19 nedd php 7.3
        #### use $phpneeded php version nedeed in _admin
        #### set alert / need php version to $phpneeded
## [1.16] - 2020/12/15
    ### Change
        #### admin.php: del last space in __('unable ') //class dcLatestVersionsLightAdmin, function adminDashboardItems
        #### langs: add french translates
## [1.15] - 2020/12/15
    ### Change
        #### admin.php: alert if php version < 7.0.0
## [1.14] - 2020/10/15
	### Change 
		#### admin.php:		workspace 'dashboard' to old name : dcLatestVersionsLight ---- like in _define.php since version 1.1
		#### install.php:	user pref dclv_show_on_dashboard  ----  like in _admin.php since version 1.1 to version 1.12
		#### install.php, admin.php: user pref dcLatestVersionsLightItems to dclv_show_on_dashboard   ----  like in _admin.php since version 1.1 to version 1.12
		#### debug.php to reflect show result
	### Delete
		#### install.php:	user pref dashboard->dcLatestVersionsLightItems, after get old value
		#### install.php:	user pref dashboard->dclv_show_on_dashboard

## [1.13] - 2020/10/02
	### Change user preferences mode in _admin.php to work with Dotclear dashboard put & get
		- worspace = dashboard
	### Change
		#### debug.php
			- no traduction for lines like __('<li><...
		#### _admin.php
			change variable $pref_show : 'dclv_show_on_dashboard' to 'dcLatestVersionsLightItems'
			change all references at $workspace to dashboard
			change return echo: add plugin name & version
			modify result in function adminDashboardItems (text show in dashboard)
		#### langs
			- ligt modify in translations help files

## [1.12] - 2020-09-20
	### changed
		#### tested on Dotclear 2.9, 2.10, 2.11, 2.12 & 2.13 with php 5.6.40 needed, no php 7.x
		#### tested on Dotclear 2.14 & 2.15 with php 7.3.5 needed, no php 7.4.x
		#### tested on Dotclear >= 2.16 & 2.17 with php 7.4.1
		#### _define.php
			- 'requires' 'core', '2.9' //tested on with php 5.6.40
		#### _install.php
			- $dc_min = '2.15'
		#### index.php
			- $dc_min = '2.15'
			- <head>: dcPage::jsPageTabs($default_tab) to dcPage::jsPageTabs($tab)
			- dcPage::breadcrumb
	###
		#### add end lines in en/help/help.html

## [1.1.1] - 2020-09-20
	### Added
		#### _defined.php 
			- url support & details
	### Changed
		#### _defined.pphp
			- 'requires': 'core', '2.15'
	### Fixed
		#### _install.php
			- name pref iem $pref_prefix .$pref_show to $pref_show
	### removed
		- files aboutlanguages.txt

## [1.1] - 2020-09-17
	### Added
		#### help files (fr & en)
		#### constant & debug files
		#### small icon
	### Changed
		#### _admin.php
			- behavior dcLatestVersionsLightAdmin

## [0.1] - Nan'Art - 2020-09 not published
	### Added
		#### index.php
			- show link to user pref dashboard
			* Variables
				- dc min version 2.16 (restriction)
				-- php min version 5.6.40 (information)
				- module_id: 'dcLatestVersionsLight';
				- workspace: 'dcLatestVersionsLight';
				- preference item: 'dclv_show_on_dashboard';
				- module_setting = array -- 'versions_type', "List of Dotclear's builds", 'stable,unstable,testing,sexy', 'string'
		#### _defined.php : 
			* 'requires': [['core', '2.16']]
			* 'priority': 2000
			* 'settings': 'pref' => '#user-favorites.dcLatestVersionsLight'
		#### _install.php
			* Variables
				- dc min version 2.16 (restriction)
				-- php min version 5.6.40 (information)
				- module_id: 'dcLatestVersionsLight';
				- workspace: 'dcLatestVersionsLight';
				- preference item: 'dclv_show_on_dashboard';
	### Changed
		- rename to version "light": dcLatestVersionsLight
		- fr translations
		- icons
	### Removed
		- no more widget
		- no more settings
	### fixed
		- _admin.php, boucle foreach (return; to continue;)

## report changelogs in old module dcLatestVersions
===================================================
2015-03-11 - Pierre Van Glabeke
* ajout mode hors ligne et codage titre widget (widget)
* encodage UNIX (LF)
* modif url de support
* modifs localisation
* suppression icône

2013.11.18 - Jean-Christian Denis
* First hg release

2013.07.22 - Jean-Christian Denis
* First release


debug.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
#
# This file is part of dcLatestVersionsLight, a plugin for Dotclear 2.
# 
# Copyright (c) 2020 Nan'Art and contributors
# a light corrected and updated version of dcLatestVersions 2015-03-11, Jean-Christian Denis and contributors
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
#
# -- END LICENSE BLOCK ------------------------------------
/**
 * @brief dcLatestVersionsLight, a plugin for Dotclear 2
 * 
 * define plugin' debug part:
	- just give some tests
	- see constant in install
 */
/* dcLatestVersionsLight/debug.php */

if (!defined('DC_CONTEXT_ADMIN')) {return;}
if(!defined('PLUG_DEBUG') || !PLUG_DEBUG) { return; }

		#user pref
			$core->auth->user_prefs->addWorkspace($workspace);
		#don't show on dashboard
			$on = $core->auth->user_prefs->$workspace->$pref_show;
		#versions types
			// $builds = 'stable,unstable,testing,sexy';
			$builds = $module_setting[0][2];
			$builds = explode(',', $builds);

#debug mode
		echo '<p class="as_h3">DEBUG mode</p>';
		#builds
			echo printR($builds, 'versions types');
		#compose
			$text = '<li><a href="%u" title="Download Dotclear version %v"> %r </a> : %v</li>';
			$li = array();
		#no versions ?
			$cache_versions = DC_TPL_CACHE.'/versions';
			$exists = file_exists($cache_versions);
			$cache_exists = $exists ?__('exists') :__('dont exists!');
			if($exists) {
				$empty = sizeof(scandir($cache_versions))<=2;
			} else {
				$error = sprintf(__('your cache versions dir %s dont exists ! Test %s.'), $cache_versions, __('force dotclear update'));
				dcPage::addErrorNotice($error);
				return;
			}
			$nb_files = isset($empty) ?sizeof(scandir($cache_versions))-2 :0;
			$cache_empty = $nb_files ?__('as files') :__('as no file!!');
			$files = $nb_files > 0 ?array_diff(scandir($cache_versions), array('..', '.')) :'';
		#infos cache
			echo '<p class="info">' .sprintf(__('Cache versions %s %s: (%s)'),$cache_versions, $cache_empty, $nb_files) .'</p>';
			if($files) { 
				echo printR($files, __('cache versions content')); 
			} else {
				$error = sprintf(__('There is no file in your cache versions dir %s. Test %s.'), $cache_versions, __('force dotclear update'));
				dcPage::addErrorNotice($error);
				return;
			}
		#witch versions type
		foreach($builds as $build) {
			$build = strtolower(trim($build));
				if (empty($build)) {
					continue;
				}
			$updater = new dcUpdate(
				DC_UPDATE_URL,
				'dotclear',
				$build,
				DC_TPL_CACHE.'/versions'
			);

			#no update for $build
			if (!$updater->check($build)) {
				continue;
			}

			$li[] = str_replace(
				array(
					'%r',
					'%v',
					'%u'
				),
				array(
					$build,
					$updater->getVersion(),
					$updater->getFileURL()
				),

				$text
			);
		}//foreach

if($li) {
		# Display
		$items[] =
		'<div class="box small" id="udclatestversionsitems">'.
		'<h3>' .'<img src="' . dcPage::getPF("$module_id/icon-small.png") . '" alt="" /> ' .html::escapeHTML(__("Dotclear's latest versions")).'</h3>'.
		'<ul>'.implode('', $li).'</ul>'.
		'</div>';
}//if li
	if(isset($items)) { echo printR($items[0], __('what should be nearly displayed on dashboard')); }


	/**
	 * usage: @ignore ?
	 * Return print_r code
	 * @param	$value			<b>mixed</b> 	-- value to print_r
	 * @param	$title			<b>string</b> 	-- title data name to show, default empty
	 * 
	 * @return  $message		<b>string</b>	-- print_r or Dotclear msg notice
	 */
function printR($value, $title ='') {
		Global $core;
		#backtrace
			$call['file'] = basename(debug_backtrace()[0]['file']);
			$call['line'] = basename(debug_backtrace()[0]['line']);
		#id
			$id = $call['line'];
			$link = "<a href='#$id'>go to $id</a>" ;
		#message
			$message = '';
			$message .= "<b>$title</b>";
			$message .= ' -- ' .__('from') .': ' .$call['file'] .' | ' .$call['line'] .' ';
			$message = '<hr>' .'<p class="as_h4" id="' .$call['line'] .'">' .$message;
			$message .= print_r($value, true);
			$message .= '<hr>';

		return $message;
}//printR


index.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
#
# This file is part of dcLatestVersionsLight, a plugin for Dotclear 2.
# 
# Copyright (c) 2020 Nan'Art and contributors
# a light corrected and updated version of dcLatestVersions 2015-03-11, Jean-Christian Denis and contributors
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
#
# -- END LICENSE BLOCK ------------------------------------
/**
 * @brief dcLatestVersionsLight, a plugin for Dotclear 2
 * 
 * define plugin' admin page:
	- just give user pref (+ url)
 */
/* dcLatestVersionsLight/index.php */

if (!defined('DC_CONTEXT_ADMIN')) {return;}

#debug ?
/*/
	define("PLUG_DEBUG", true);
//*/

#@ignore information about Dotclear & php versions
	#dc min version (tested)
		$dc_best = '2.15';	//php 7.3.5
		// $dc_best = '2.16';	//php 7.4.1
	#php min version (tested)
		$php_min = '5.6.40';	//dc version 2.9 to 2.13
		$php_min = '7.3.5';		//dc version 2.14 to 2.15
		$php_min = '7.4.1'; 	//dc version 2.16+

/*------please, dont change bellow---------*/
#module id
	$module_id = dcLatestVersionsLightAdmin::$module_id; //dcLatestVersionsLight
#module_setting
	$module_setting = [
		[	'versions_type',
			"List of Dotclear's builds",
			'stable,unstable,testing,sexy', //sexy ignore?
			'string'
		]
	];
#user prefs
	$workspace		= dcLatestVersionsLightAdmin::$workspace; //'dcLatestVersionsLight';
	$pref_show		= dcLatestVersionsLightAdmin::$pref_show; //'dclv_show_on_dashboard';

#page
	$index_title 	= __('Plugin informations');
	$index_comment 	= sprintf(__("plugin %s information"), __($module_id));

#preferences
		$core->auth->user_prefs->addWorkspace($workspace);
		$show 		= $core->auth->user_prefs->$workspace->$pref_show;
		$show_on 	= $show ?__('show on dashboard.') :__('not display on dashboard.');

#versions types
$version_types	= dcLatestVersionsLightAdmin::$version_types; //'stable,unstable,testing,sexy';

#tab
$tab  = empty($_REQUEST['tab']) ? '' : $_REQUEST['tab'];

?>
<html>
<head>
	<title><?php echo $module_id; ?></title>
	<!-- onglets -->
	<?php 
	
	echo dcPage::jsPageTabs($tab); 
	?>
</head>
<body>
 <?php
# menu
    echo dcPage::breadcrumb(
        array(
            __('Plugins')     => '',
            __($module_id) => ''
        )
    );
# messages
	echo dcPage::notices();
?>

    <div class="fieldset">
	<p class="as_h3"><?php echo $index_comment; ?></p>
	<p class="as_h4"><?php printf(__("your current %s status is '<strong>%s</strong>'"), __('preferences about dotclear updates'), $show_on); ?></p>
	<p class="info">
		<?php 
		#add '.dcLatestVersionsLight' to link id version >= 2.15
				$id = version_compare($dc_best, DC_VERSION, '>=') ?'' :'.dcLatestVersionsLight';
			echo sprintf(
					__('You can change this options in your %s %s.'),
					'<a href="' . $core->adminurl->get('admin.user.preferences') 
								. '#user-favorites' .$id .'">' 
								. __('Preferences on Dashboard') . '</a>',
					'<br><i>' .__('also see below: '). __('Plugin settings (in blog parameters)') .'</i>'
						);
		?>
	</p>
	<hr/>
	<p class="info">
		<?php
			$versions		= sprintf(__('Your Dotclear version is %s - %s.'),DC_VERSION,  DC_UPDATE_VERSION );
			$versions		.= ' | ';
			$versions		.= sprintf(__('Your php version is %s.'),phpversion() );
			echo $versions;
		?>
	<p>

<?php
/* just if PLUG_DEBUG */
if(defined('PLUG_DEBUG') && PLUG_DEBUG) {
	$file = dirname(__FILE__) .'/debug.php';
		if(file_exists($file)) {
			require_once $file;
		}
}//if PLUG_DEBUG
echo '</div>';
/* help */
dcPage::helpBlock('dcLatestVersionsLight');
?>

</body>
</html>


_admin.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
#
# This file is part of dcLatestVersionsLight, a plugin for Dotclear 2.
# 
# Copyright (c) 2020 Nan'Art and contributors
# a light corrected and updated version of dcLatestVersions 2015-03-11, Jean-Christian Denis and contributors
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
#
# -- END LICENSE BLOCK ------------------------------------
/* dcLatestVersionsLight/_admin.php */

if (!defined('DC_RC_PATH')) { return; }
// Public and Admin mode

if (!defined('DC_CONTEXT_ADMIN')) { return; }
// Admin mode only

// dead but useful code, in order to have translations
__('dcLatestVersionsLight').__('Show the latest available versions of Dotclear in dashboard');

#plugin title
$module_id = 'dcLatestVersionsLight';

# Sidebar menu
$_menu['Plugins']->addItem(
	__($module_id),
	$core->adminurl->get('admin.plugin.' .$module_id),
	dcPage::getPF($module_id .'/icon.png'),
	preg_match('/' . preg_quote($core->adminurl->get('admin.plugin.' .$module_id)) . '(&.*)?$/', $_SERVER['REQUEST_URI']),
	$core->auth->check('admin', $core->blog->id)
);

/* behaviors */
$core->addBehavior('adminDashboardFavorites',array('dcLatestVersionsLightAdmin','adminDashboardFavorites'));
$core->addBehavior('adminDashboardOptionsForm',array('dcLatestVersionsLightAdmin','adminDashboardOptionsForm'));
$core->addBehavior('adminAfterDashboardOptionsUpdate',array('dcLatestVersionsLightAdmin','adminAfterDashboardOptionsUpdate'));
$core->addBehavior('adminDashboardItems',array('dcLatestVersionsLightAdmin','adminDashboardItems'));

class dcLatestVersionsLightAdmin
{
	public static	$module_id		= 'dcLatestVersionsLight';
	public static 	$workspace		= 'dcLatestVersionsLight';
	public static	$pref_show		= 'dclv_show_on_dashboard';
	public static	$version_types	= 'stable,unstable,testing,sexy'; //sexy ignore?
	public static	$versions_cache	= '/versions';

    /**
     * Favorites.
     *
     * @param    $core    <b>dcCore</b>    dcCore instance
     * @param    $favs    <b>arrayObject</b>    Array of favs
     */
	public static function adminDashboardFavorites($core, $favs)
	{
		$class = __CLASS__;
		$plugin_name = self::$module_id;
		
		$favs->register('dcLatestVersionsLight', [
			'title'		=> __('Dotclear latests versions light'),
			'url'		  => $core->adminurl->get('admin.plugin.' .$plugin_name),
			'small-icon'   => dcPage::getPF($plugin_name .'/icon.png'),
			'large-icon'   => dcPage::getPF($plugin_name .'/icon-big.png'),
			'permissions'  => 'admin',
			// 'dashboard_cb' => [$class, 'adminDashboardItems']
		]);
	}//adminDashboardFavorites

	/**
	 * User preferences form.
	 *
	 * @param	$core	<b>object</b>	dcCore instance or record
	 */
	public static function adminDashboardOptionsForm($core)
	{
		$show		= 	self::$pref_show;
		$workspace	=	self::$workspace;
		#is preference
			if (!$core->auth->user_prefs->$workspace->prefExists($show)) {
				$core->auth->user_prefs->$workspace->put(
					$show,
					false,
					'boolean'
				);
			}
		#user pref
			$show_value = $core->auth->user_prefs->$workspace->get($show);

		#some plugin infos
			#plugin name
				$p_name = self::$module_id;
			#plugin version
				$p_version = $core->plugins->moduleInfo($p_name, 'version');
			#plugin message
				$p_message = sprintf(__('%s - %s'), __($p_name), $p_version);

		echo '<div class="fieldset" id="' .$workspace .'">';
		echo '<h4>' . $p_message .'</h4>';
		echo '<p>';
			echo '<label for="' .$show .'" class="classic">';
				echo form::checkbox($show, 1, $show_value);
					echo __('Show Dotclear\'s latest versions on dashboard.');
			echo '</label>';
		echo '</p>';
		echo '</div>';
	}//adminDashboardOptionsForm

	/**
	 * User preferences form get updated form
	 *
	 * @param	$user_id	<b>string</b>	user id
	 */
	public static function adminAfterDashboardOptionsUpdate($user_id)
	{
		Global $core;

		$show		= 	self::$pref_show;
		$workspace	=	self::$workspace;
		# Get and store user's prefs for plugin options 
        try {
            // Hosting monitor options
            $core->auth->user_prefs->$workspace->put($show, !empty($_POST[$show]), 'boolean');
        } catch (Exception $e) {
            $core->error->add($e->getMessage());
        }
	}//adminAfterDashboardOptionsUpdate

	/**
	 * set show on dashboard
	 *
     * @param    $core    <b>dcCore</b>    		 dcCore instance
     * @param    $items   <b>arrayObject</b>    Dashboard items
	 */
	public static function adminDashboardItems($core, $items)
	{
		$show		= 	self::$pref_show;
		$workspace	=	self::$workspace;
		#don't show on dashboard
			if (!$core->auth->user_prefs->$workspace->get($show)) { return; }

		#get version types
			$builds = self::$version_types;
			$builds = explode(',', $builds);
		
		#get path cache/verions
			$path = self::$versions_cache;
		#module id
			$module_id = self::$module_id;

		#compose
			$title		= "%r";
			$version	= sprintf(__('Dotclear %s'), "%v");
			$text		= '<li><a href="%u" title="' .$title .'">%r</a> : ' .$version .'</li>';
			$li = array();

		foreach($builds as $build) {
			$build = strtolower(trim($build));
			if (empty($build)) {
				continue;
			}

			$updater = new dcUpdate(
				DC_UPDATE_URL,
				'dotclear',
				$build,
				DC_TPL_CACHE .$path
			);

			#no update for $build
			if (!$updater->check($build)) {
				continue;
			}

			$li[] = str_replace(
				array(
					'%r',
					'%v',
					'%u'
				),
				array(
					__('Download version ') .$build,
					$updater->getVersion(),
					$updater->getFileURL()
				),
				$text
			);
		}

		if (empty($li)) { return null; }


		//some plugin infos
		#plugin name
			$p_name = self::$module_id;
		#plugin version
			$p_version = $core->plugins->moduleInfo($p_name, 'version');

        #msg
            $ok = (version_compare(PHP_VERSION, '7.0.0') >= 0) ?__('able') :__('unable');
            $type = (version_compare(PHP_VERSION, '7.0.0') >= 0) ?__('success') :__('warning');
            $msg = sprintf(__('Your PHP version %s is %s to support the future Dotclear version 2.19.'), phpversion(), $ok);

		# Display
        $items[] = new ArrayObject([
		'<div class="box small" id="udclatestversionsitems">'.
		'<h3>' .'<img src="' . dcPage::getPF("$module_id/icon-small.png") . '" alt="" /> ' .html::escapeHTML(__("Dotclear latests versions - light")) .'</h3>'.
		'<ul>'.implode('', $li).'</ul>'.
        '<p class="' .$type .'">' .$msg .'</p>'.
		'</div>'
 ]);
	}//adminDashboardItems

}//dcLatestVersionsLightAdmin


_install.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
#
# This file is part of dcLatestVersionsLight, a plugin for Dotclear 2.
# 
# Copyright (c) 2020 Nan'Art and contributors
# a light corrected and updated version of dcLatestVersions 2015-03-11, Jean-Christian Denis and contributors
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
#
# -- END LICENSE BLOCK ------------------------------------
/* dcLatestVersionsLight/_install.php */

if (!defined('DC_CONTEXT_ADMIN')) {
	return null;
}

#dc min version (tested)
	$dc_min = '2.9';
#php min version (tested)
#@ignore
	$php_min = '5.6.40';

/*------please, dont change bellow---------*/
#module id
	$module_id = 'dcLatestVersionsLight';
#user prefs
	$workspace		= 'dcLatestVersionsLight';
	$pref_show		= 'dclv_show_on_dashboard';
#versions
	$new_version = $core->plugins->moduleInfo($module_id, 'version');
	$old_version = $core->getVersion($module_id);

# -- Nothing to change below --
if (version_compare($old_version, $new_version, '>=')) {
//*/
	return null;
//*/
}

try {
	# Check Dotclear version @ignore
		if (!method_exists('dcUtils', 'versionsCompare') 
		 || dcUtils::versionsCompare(DC_VERSION, $dc_min, '<', false)) {
			throw new Exception(sprintf(
				'%s requires Dotclear %s', $module_id, $dc_min
			));
		}
	# set ws & user pref if not defined
		if (!$core->auth->user_prefs->$workspace->prefExists($pref_show)) {
			$core->auth->user_prefs->$workspace->put(
				$pref_show,
				false,
				'boolean',
				'Show Dotclear latest versions on dashboard.'
			);
		}
	#get old user prefs
		$old_pref = $core->auth->user_prefs->dashboard->prefExists('dcLatestVersionsLightItems') 
			? $core->auth->user_prefs->dashboard->get('dcLatestVersionsLightItems')
			: null;
	#get last user prefs
		$new_pref = $core->auth->user_prefs->$workspace->get($pref_show);
	#set/keep pref -- if dcLatestVersionsLightItems or dclv_show_on_dashboard
		$user_pref = ($old_pref || $new_pref) ?true :false;
		$core->auth->user_prefs->dashboard->put($pref_show, $user_pref, 'boolean', 'Show Dotclear latest versions on dashboard.');
	#del obsolete user prefs
		# suppression de la préférence
			$core->auth->user_prefs->dashboard->drop('dcLatestVersionsLightItems');
			$core->auth->user_prefs->dashboard->drop($pref_show);

	#set module version
		$core->setVersion($module_id, $new_version);

	return true;

} catch (Exception $e) {
	$core->error->add($e->getMessage());

	return false;
}


locales/en/resources.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
#
# This file is part of dcLatestVersionsLight, a plugin for Dotclear 2.
# 
# Copyright (c) 2020 Nan'Art and contributors
# a light corrected and updated version of dcLatestVersions 2015-03-11, Jean-Christian Denis and contributors
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
#
# -- END LICENSE BLOCK ------------------------------------
/**
 * @brief dcLatestVersionsLight, a plugin for Dotclear 2
 * 
 * help resources loader
 */
/* dcLatestVersionsLight/locales/en/resources.php */

if (!isset($__resources['help']['dcLatestVersionsLight'])) {
    $__resources['help']['dcLatestVersionsLight'] = dirname(__FILE__) . '/help/help.html';
}

locales/fr/main.lang.php
<?php
// Language: Français 
// Module: dcLatestVersionsLight - 1.15
// Date: 2020-12-15 15:02:58 
// Translated with dcTranslater - 2013.05.11 

#_admin.php:23
$GLOBALS['__l10n']['dcLatestVersionsLight'] = 'Dotclear dernières versions Light';

#_admin.php:23
$GLOBALS['__l10n']['Show the latest available versions of Dotclear in dashboard'] = 'Montrer les dernières versions sur le tableau de bord';

#_admin.php:63
$GLOBALS['__l10n']['Dotclear latests versions light'] = 'Dotclear dernières versions Light';

#_admin.php:105
$GLOBALS['__l10n']['Show Dotclear\'s latest versions on dashboard.'] = 'Montrer les dernières versions sur le tableau de bord';

#_admin.php:184
$GLOBALS['__l10n']['Download version'] = 'Télécharger la version';

#_admin.php:202
$GLOBALS['__l10n']['able'] = 'à même';

#_admin.php:202
$GLOBALS['__l10n']['unable'] = 'incapable';

#_admin.php:204
$GLOBALS['__l10n']['Your PHP version %s is %s to support the future Dotclear version 2.19.'] = 'Votre version PHP %s est %s de supporter la future version 2.19 de Dotclear.';

#_admin.php:209
$GLOBALS['__l10n']['Dotclear latests versions - light'] = 'Dotclear dernières versions Light';

#debug.php:45
$GLOBALS['__l10n']['exists'] = 'existe';

#debug.php:45
$GLOBALS['__l10n']['dont exists!'] = 'n\'existe pas';

#debug.php:49
$GLOBALS['__l10n']['your cache versions dir %s dont exists ! Test %s.'] = 'Le répertoire cache des versions %s n\'existe pas ! Essayer de %s';

#debug.php:49
#debug.php:61
$GLOBALS['__l10n']['force dotclear update'] = 'forcer les mises à jour de Dotclear';

#debug.php:54
$GLOBALS['__l10n']['as files'] = 'contient des fichiers';

#debug.php:54
$GLOBALS['__l10n']['as no file!!'] = 'ne contient aucun fichier';

#debug.php:59
$GLOBALS['__l10n']['cache versions content'] = 'contenu du cache versions';

#debug.php:61
$GLOBALS['__l10n']['There is no file in your cache versions dir %s. Test %s.'] = 'Il n\'y a aucun fichier dans le répertoire cache versions %s. Essayer de %s';

#debug.php:103
$GLOBALS['__l10n']['Dotclear\'s latest versions'] = 'Dernières versions Dotclear';

#debug.php:107
$GLOBALS['__l10n']['what should be nearly displayed on dashboard'] = 'Ce qui devrait approximativement être affiché sur le tableau de bord';

#debug.php:129
$GLOBALS['__l10n']['from'] = 'de';

#index.php:54
$GLOBALS['__l10n']['Plugin informations'] = 'Informations plugin';

#index.php:55
$GLOBALS['__l10n']['plugin %s information'] = 'informationsplugin %s';

#index.php:60
$GLOBALS['__l10n']['show on dashboard.'] = 'montrer sur le tableau de bord';

#index.php:60
$GLOBALS['__l10n']['not display on dashboard.'] = 'ne pas afficher sur le tableau de bord';

#index.php:93
$GLOBALS['__l10n']['your current %s status is \'<strong>%s</strong>\''] = 'Votre statut actuel %s est \'<strong>%s</strong>\'';

#index.php:93
$GLOBALS['__l10n']['preferences about dotclear updates'] = 'préférences à propos des mises à jour de Dotclear';

#index.php:99
$GLOBALS['__l10n']['You can change this options in your %s %s.'] = 'Vous pouvez modifier ces options dans vos %s %s.';

#index.php:102
$GLOBALS['__l10n']['Preferences on Dashboard'] = 'Préférences sur le tableau de bord';

#index.php:103
$GLOBALS['__l10n']['also see below:'] = 'voir aussi plus bas:';

#index.php:110
$GLOBALS['__l10n']['Your Dotclear version is %s - %s.'] = 'Votre version Dotclear :  %s - %s.';

#index.php:112
$GLOBALS['__l10n']['Your php version is %s.'] = 'Votre version php : %s.';

?>


_define.php
<?php
# -- BEGIN LICENSE BLOCK ----------------------------------
#
# This file is part of dcLatestVersionsLight, a plugin for Dotclear 2.
# 
# Copyright (c) 2020 Nan'Art and contributors
# a light corrected and updated version of dcLatestVersions 2015-03-11, Jean-Christian Denis and contributors
# 
# Licensed under the GPL version 2.0 license.
# A copy of this license is available in LICENSE file or at
# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
#
# -- END LICENSE BLOCK ------------------------------------
/* dcLatestVersionsLight/_define.php */

if (!defined('DC_RC_PATH')) {
	return null;
}
 
$this->registerModule(
	/* Name */
	"dcLatestVersionsLight",
	/* Description*/
	"Show the latest available versions of Dotclear in dashboard",
	/* Author */
	"Nan'Art and contributors",
	/* Version */
	'1.17',
	/* Properties */
	[
		'requires'    => [['core', '2.9']],				// dotclear min version (tested)
		'permissions' => 'contentadmin',					// Permissions all admin dashboard user
		'type' => 'plugin',									// type
		'priority' => 		2000,							//priority
		'support' => 'https://github.com/nanart59/dcLatestVersionsLight',
		'details' => 'https://github.com/nanart59/dcLatestVersionsLight',
		'settings'    => [									// Settings
			'pref' => '#user-favorites.dcLatestVersionsLight'
			],
	]
);
